contribute/architecture/backend/package-hierarchy.md

```diff
@@ -0,0 +1,9 @@
+# Package hierarchy
+
+The Go package hierarchy in Grafana should be organized logically (Ben Johnson's
+[article](https://medium.com/@benbjohnson/standard-package-layout-7cdbc8391fc1) served as inspiration), according to the 
+following principles:
+
+* Domain types and interfaces should be in "root" packages (not necessarily at the very top, of the hierarchy, but
+  logical roots)
+* Sub-packages should depend on roots - sub-packages here typically contain implementations, for example of services
```

Comment:
I think it would be great if we could add a clear example here - might help paint the picture more. WDYT?

Reply:
@wbrowne sounds good to me.

Reply:
Will do it, but when time permits.

Reply:
@wbrowne done.